docs/content/en/docs/crd-ref/TaskDefinition/index.md

```diff
@@ -0,0 +1,279 @@
+  for more information about the runtime.
+  In the future, we intend to support additional runtimes,
+  especially running a container image directly.
+
+  The `function` can be defined as one of the following:
+
+  * **inline** - Include the actual executable code to execute.
+    This can be written as a full-fledged Deno script
+    that is included in this file.
+    For example:
+
+    ```function:
+    inline:
+      code: |
+        console.log("Deployment Task has been executed");
+    ```
+
+  * **httpRef** - Specify a Deno script to be executed at runtime
+    from the remote webserver that is specified.
+    For example:
+
+    ```yaml
+    name: hello-keptn-http
+      spec:
+        function:
+          httpRef:
+            url: <url>
+    ```
+
+  * **functionRef** -- Execute another `KeptnTaskDefinition` that has been defined.
+    Populate this field with the value of the `name` field
+    for the `KeptnTaskDefinition` to be called.
+    This is commonly used to call a general function
+    that is used in multiple places, possibly with different parameters.
+    An example is:
+
+     ```yaml
+     spec:
+       function:
+         functionRef:
+           name: slack-notification
+     ```
+
+    This can also be used to group a set of tasks
+    into a single `KeptnTaskDefinitions`,
+    such as defining a `KeptnTaskDefinition` for testing.
+    In this case, it other, existing `KeptnTaskDefinition`s
+    for each type of test to be run,
+    specifying each by the value of the `name` field.
+
+* **parameters** - An optional field to supply input parameters to a function.
+  The Lifecycle Toolkit passes the values defined inside the `map` field
+  as a JSON object.
+  For example:
+
+   ```spec:
+       parameters:
+         map:
+           textMessage: "This is my configuration"
+   ```
+
+   The JSON object can be read
+   through the `DATA` environment variable using `Deno.env.get("DATA");`.
+
+   Multi-level maps are not supported at this time.
+
+   Currently only one secret can be passed.
+   The secret must have a `key` called `SECURE_DATA`.
+   It can be accessed via the environment variable `Deno.env.get("SECURE_DATA")`.
+   See [Context](#context) for details.
+
+* **secureParameters** -- An optional field used to pass a Kubernetes secret.
+  The `secret` value is the Kubernetes secret name
+  that is mounted into the runtime and made available to functions
+  using the `SECURE_DATA` environment variable.
+  For example:
+
+  ```yaml
+      secureParameters:
+        secret: slack-token
+   ```
+
+   See [Create secret text](#create-secret-text) for details.
+
+## Usage
+
+A Task is responsible for executing the TaskDefinition of a workload.
+The execution is done by spawning a Kubernetes Job to handle a single Task.
+In its state, it tracks the current status of this Kubernetes Job.
+
+### Context
+
+A context environment variable is available via `Deno.env.get("CONTEXT")`.
```

Comment:
If the content of the context variable is defined by us, we should also include that in the reference above.
Especially, what data the user actually has available in the context.